w1r3/infra/metrics/main.tf

```diff
@@ -0,0 +1,222 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+variable "metric-prefix" {
+    default = "ssb/w1r3"
+}
+
+resource "google_monitoring_metric_descriptor" "latency" {
+  description = "Operation latency as measured by the benchmark."
+  display_name = "Operation Latency"
+  type = "workload.googleapis.com/${var.metric-prefix}/latency"
+  metric_kind = "CUMULATIVE"
+  value_type = "DISTRIBUTION"
+  unit = "s"
+
+  labels {
+    key = "service_namespace"
+    value_type = "STRING"
+  }
+  labels {
+    key = "service_name"
+    value_type = "STRING"
+  }
+  labels {
+    key = "service_instance_id"
```

Comment:
What is the "service"?

Reply:
In this case, `w1r3`.  We are using the `generic_task` monitored resource:

https://cloud.google.com/monitoring/api/resources#tag_generic_task

All the programs will call themselves `w1r3`, and assign a UUID to each instance.